### DISCUSSION TITLE
Lately the weather fetch has been wrong

### DISCUSSION THREAD

**Discussion by stvzldn** [mytourbook/mytourbook#1423] [General]
Wrong weather is especially obvious when the fetch reports rain but the sky is sunny and we are drought dry - like today's ride.

![Weather fetch wrong](https://github.com/user-attachments/assets/3580c273-c253-4e4d-a197-c6a5efb56563)

**Comment by cyc1ingsir** [comment]
I made similar observations lately but put it down to the weather data provider.
Just an example from last Sunday. I couldn't have been more sunny:
![image](https://github.com/user-attachments/assets/8bec22f4-ffe4-405e-a44c-30493e235524)
Pictures taken on the ride
![image](https://github.com/user-attachments/assets/a036329a-f873-407d-ae39-ba8ba3f07c1f)
and weather data gathered from Strava
![image](https://github.com/user-attachments/assets/71a027d5-6809-4fd7-bf23-7d14e3b639a4)

**Comment by FJBDev** [comment]
> Wrong weather is especially obvious when the fetch reports rain but the sky is sunny and we are drought dry - like today's ride.
> 

Did  you try and compare another provider like ? MT has 3 weather providers

**Comment by Thompson-ongithub** [comment]
I had the same phenomenon, but put it down to the weather provider, since Weather API got it right, whilst OpenWeatherMap (which I usually prefer) showed a completely wrong weather.

**Comment by stvzldn** [comment]
OpenWeatherMap produced the rain hallucination in my image above.  WeatherAPI was better but still called a clear blue sky as overcast but at least reported no non-existant rain.

**Comment by cyc1ingsir** [comment]
I have the provider set to OpenWeatherMap usally, since it allows to retrieve data for tours further in the past. This had proven especially useful when importing the tours from my bike packing holiday.
Retrieving the weather data for the example tour above with Weather API, it looks quite realistic. The wind direction doesn't fit, however.
